### PR TITLE
Print the full qualification of a type.

### PR DESCRIPTION
**Pull request by vgvassilev**

### CODE REVIEW

**Comment by dpiparo** [comment]
Hi @vgvassilev , is this https://sft.its.cern.ch/jira/browse/ROOT-8897 ?

**Comment by vgvassilev** [comment]
@dpiparo looks like...

**Comment by bluehood** [comment]
that issue is weirder than the title suggests: the following works fine:
```c++
root [0] ROOT::Experimental::TDF::TInterface<ROOT::Detail::TDF::TFilterBase> c2 = d.Filter([]() { return true; })
```
but this, with `auto` instead of the full name typed out, doesn't:
```c++
auto c = d.Filter([]() { return true; }
```

maybe we could add a test?

**Comment by vgvassilev** [comment]
I think I know where the problem is. I have a test that triggers it as part of this PR.

File: interpreter/cling/lib/Utils/AST.cpp [hunk 1]
```diff
@@ -1693,6 +1693,7 @@ namespace utils {
     PrintingPolicy Policy(Ctx.getPrintingPolicy());
     Policy.SuppressScope = false;
     Policy.AnonymousTagLocations = false;
+    Policy.FullyQualifiedName = true;
```
Comment (pcanal) [inline interpreter/cling/lib/Utils/AST.cpp]: Please make sure that this does not hide a missing feature in GetFullyQualifiedType which is suppose to already have built-in the full qualification.

**Comment by Axel-Naumann** [comment]
@vgvassilev how did you address the comment https://github.com/root-project/root/pull/1787#pullrequestreview-106971834 by @pcanal:
> Please make sure that this does not hide a missing feature in GetFullyQualifiedType which is supposed to already have built-in the full qualification.

I am collecting a set of tests (failures) of the clang version of `GetFullyQualifiedName`. Adding "your cases" to that list might be a way of addressing Philippe's comment. Do you have such test cases?

**Comment by Axel-Naumann** [comment]
Hmm I guess the addition to the test suite qualifies for that. Thanks, settled!

**Pull request merged**